### PR TITLE
Niloofar / Tooltip component 

### PR DESCRIPTION
**Pull request by niloofar-deriv**
Implemented the `Tooltip` component

Screenshot 2023-10-30 at 10 54 27 AM

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -40,17 +40,6 @@ module.exports = {
                 'gradient-success': 'linear-gradient(to top, #ffffff, rgba(75, 180, 179, 0.16))',
                 'gradient-danger': 'linear-gradient(to top, #ffffff, rgba(255, 68, 79, 0.16))',
             },
-            fontSize: {
-                base: '1rem',
```
Comment (nada-deriv) [inline tailwind.config.js:44]: i think i used the text-base somewhere in my pr. 😅

Comment (niloofar-deriv) [reply tailwind.config.js:44]: This breaks everything I had to remove it and use default font sizes

File: src/components/common/tooltip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    asChild?: boolean;
+};
+
+const Tooltip = ({ content, children, asChild }: PropsWithChildren<TTooltip>) => (
+    <Provider>
+        <Root>
+            <Trigger asChild={asChild}>{children}</Trigger>
+            <Portal>
+                <Content
+                    className='data-[state=delayed-open]:data-[side=top]:animate-slideDownAndFade data-[state=delayed-open]:data-[side=right]:animate-slideLeftAndFade data-[state=delayed-open]:data-[side=left]:animate-slideRightAndFade data-[state=delayed-open]:data-[side=bottom]:animate-slideUpAndFade rounded-[4px] bg-active p-1 text-xs text-prominent'
```
Comment (balakrishna-deriv) [inline src/components/common/tooltip/index.tsx]: @niloofar-deriv I'm just curious. Where do we get these styles? 😄

Comment (niloofar-deriv) [reply src/components/common/tooltip/index.tsx]: Browser's inspect 😂

**Review by balakrishna-deriv — APPROVED** [review]

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**